### PR TITLE
test: fix some linting issues

### PR DESCRIPTION
**Pull request by vayan**
Starting to fix some linting issues to re-enable the CI linter soon.

This is a first pass, still more to work on :)

### CODE REVIEW

File: src/dismantle/extension/extensions.py [hunk 1]
```diff
@@ -30,25 +31,18 @@ def __init__(self, types, packages, prefix) -> None:
...
                     files[:] = [f for f in files if f.endswith(self._include)]
                     for name in files:
                         path = Path(root, name)
-                        if len(path.suffixes) >= 2:
-                            n_length = len(path.stem) - len(path.suffixes[0])
-                            ext_name = path.stem[:n_length]
-                        else:
-                            ext_name = path.stem
-                        prefix = _prefix + '.extension.' + ext_name
+                        # remove the last two prefixes (e.g .cpython-37.pyc)
```
Comment (garysb) [inline src/dismantle/extension/extensions.py:44]: Please confirm this works with a module based (nested directory using an __init__ file) and a file based module.

File: tests/extensions_class_test.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
+import os
+import shutil
+import tempfile
+import unittest
```
Comment (garysb) [inline tests/extensions_class_test.py:4]: Is there a reason we are switching away from pytest to use unitest instead?

**Review by garysb — CHANGES_REQUESTED** [review]

**Comment by rakhra** [comment]
not sure but this could be helpful:
- there is a `pep8` formatter - https://github.com/hhatto/autopep8
- there is a git hub action to trigger it - https://github.com/marketplace/actions/autopep8

**Comment by dfundingsland** [comment]
Thanks @rakhra. There's always black, which is likely the most popular `pep8` formatter, though not without it's issues.

- https://github.com/psf/black

File: tests/extensions_class_test.py (hunk 2 above)
Comment (vayan) [reply tests/extensions_class_test.py:4]: just a preference, they both work well together anyway. 

I like the `setUp` / `tearDown` of unittest

**Comment by vayan** [comment]
there's a linter in this project, but there's already some issues on `main` that's why I disabled the GitHub Action for it until I fix them.

File: tests/extensions_class_test.py (hunk 2 above)
Comment (garysb) [reply tests/extensions_class_test.py:4]: Sure, but we then have another dependancy and another module requirement.

Comment (vayan) [reply tests/extensions_class_test.py:4]: it comes packaged with Python afaik

Comment (dfundingsland) [reply tests/extensions_class_test.py:4]: Yes, `unittest` is packaged with Python. `pytest` is, essentially, just a 3rd party unittest framework, with the intention to make writing tests easier.

File: src/dismantle/extension/extensions.py (hunk 1 above)
Comment (vayan) [reply src/dismantle/extension/extensions.py:44]: Yeah I haven't seen any changes in behavior

File: tests/extensions_class_test.py (hunk 2 above)
Comment (garysb) [reply tests/extensions_class_test.py:4]: Ok, so we are switching to unittest away from pytest? I just don't want to make a mess and have multiple test suites. The reason I redid my original unittest tests and switched to pytest was that its much easier to understand, especially when working with things like emulating a web api and for using fixtures.

Comment (vayan) [reply tests/extensions_class_test.py:4]: Oh it's the opposite for me, I find `pytest` fixtures very confusing. 

I've used `unittest` in `devkit`(https://github.com/area28technologies/devkit/blob/master/tests/build_test.py) and I find stuff like `f'{os.path.dirname(__file__)}/fixtures/fake-pkg'` more clear than some decorators and some hidden magic with folder names.

But we could do it in `pytest` too.

And for emulating API it's not related to the test suite. I've used https://github.com/getsentry/responses in devkit seems easier and more used than https://github.com/csernazs/pytest-httpserver 

I actually don't really mind either, I'll just rewrite them in `pytest`  👍

**Review by garysb — APPROVED** [review]

**Review by dfundingsland — APPROVED** [review]

**Pull request merged**